src/cascadia/TerminalApp/AppLogic.cpp

```diff
@@ -859,59 +864,29 @@ namespace winrt::TerminalApp::implementation
     {
-        // Get the containing folder.
         const std::filesystem::path settingsPath{ std::wstring_view{ CascadiaSettings::SettingsPath() } };
-        const auto folder = settingsPath.parent_path();
-
-        _reader.create(folder.c_str(),
-                       false,
-                       wil::FolderChangeEvents::All,
-                       [this, settingsPath](wil::FolderChangeEvent event, PCWSTR fileModified) {
-                           // We want file modifications, AND when files are renamed to be
-                           // settings.json. This second case will oftentimes happen with text
-                           // editors, who will write a temp file, then rename it to be the
-                           // actual file you wrote. So listen for that too.
-                           if (!(event == wil::FolderChangeEvent::Modified ||
-                                 event == wil::FolderChangeEvent::RenameNewName ||
-                                 event == wil::FolderChangeEvent::Removed))
-                           {
-                               return;
-                           }
-
-                           std::filesystem::path modifiedFilePath = fileModified;
-
-                           // Getting basename (filename.ext)
-                           const auto settingsBasename = settingsPath.filename();
-                           const auto modifiedBasename = modifiedFilePath.filename();
-
-                           if (settingsBasename == modifiedBasename)
-                           {
-                               this->_DispatchReloadSettings();
-                           }
-                       });
-    }
-
-    // Method Description:
-    // - Dispatches a settings reload with debounce.
-    //   Text editors implement Save in a bunch of different ways, so
-    //   this stops us from reloading too many times or too quickly.
-    fire_and_forget AppLogic::_DispatchReloadSettings()
-    {
-        if (_settingsReloadQueued.exchange(true))
-        {
-            co_return;
-        }
-
-        auto weakSelf = get_weak();
-
-        co_await winrt::resume_after(std::chrono::milliseconds(100));
-        co_await winrt::resume_foreground(_root->Dispatcher());
-
-        if (auto self{ weakSelf.get() })
-        {
-            _ReloadSettings();
-            _settingsReloadQueued.store(false);
-        }
+        const std::filesystem::path statePath{ std::wstring_view{ ApplicationState::SharedInstance().FilePath() } };
+
+        _reader.create(
+            settingsPath.parent_path().c_str(),
+            false,
+            // We want file modifications, AND when files are renamed to be
+            // settings.json. This second case will oftentimes happen with text
+            // editors, who will write a temp file, then rename it to be the
+            // actual file you wrote. So listen for that too.
+            wil::FolderChangeEvents::FileName | wil::FolderChangeEvents::LastWriteTime,

```

Comment:
We previously listened for `wil::FolderChangeEvents::All` and then filtered the events, but this isn't necessary...
* `FileName` maps to `FILE_NOTIFY_CHANGE_FILE_NAME` and includes renaming, creating, or deleting a file name.
* `LastWriteTime` maps to `FILE_NOTIFY_CHANGE_LAST_WRITE` and includes any writes that are flushed to disk.